x-pack/plugins/security_solution/server/lib/detection_engine/rule_response_actions/endpoint_response_action.ts

```diff
@@ -29,20 +33,54 @@ export const endpointResponseAction = (
     command,
     rule_id: alerts[0][ALERT_RULE_UUID],
     rule_name: alerts[0][ALERT_RULE_NAME],
-    agent_type: 'endpoint' as const,
   };
 
   if (isIsolateAction(responseAction.params)) {
-    const alertsPerAgent = getIsolateAlerts(alerts);
-    each(alertsPerAgent, (actionPayload) => {
-      return endpointAppContextService.getActionCreateService().createActionFromAlert(
-        {
-          ...actionPayload,
-          ...commonData,
-        },
-        actionPayload.endpoint_ids
-      );
-    });
+    if (request) {
+      const casesClient = await endpointAppContextService.getCasesClient(request);
+      const connectorActions = await endpointAppContextService.getActionsClientWithRequest(request);
+
+      each(alerts, async (alert) => {
+        const agentType = ['sentinel_one', 'sentinel_one_cloud_funnel'].includes(alert.event.module)
+          ? 'sentinel_one'
+          : 'endpoint';
+        const endpointId =
+          agentType === 'sentinel_one'
+            ? alert.observer?.serial_number ?? alert.sentinel_one_cloud_funnel?.event?.agent?.uuid
+            : alert.agent.id;
+        const user = endpointAppContextService.security?.authc.getCurrentUser(request);
+
+        const responseActionsClient: ResponseActionsClient = getResponseActionsClient(
```

Comment:
Endpoint system action finds a proper `responseActionsClient` to use to execute the response action